sw/device/silicon_creator/lib/sigverify_tests/sigverify_set_testvectors.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import sys
+
+import hjson
+from Crypto.Hash import SHA256
+from mako.template import Template
+
+'''
+Read in a JSON test vector file, convert the test vector to C constants, and
+generate a header file with these test vectors.
+'''
+
+# Number of 32-bit words in a 3072-bit number
+RSA_3072_NUMWORDS = int(3072 / 32)
+
+# Number of 32-bit words in a 256-bit number
+INT_256_NUMWORDS = int(256 / 32)
+
+# Template file name
+TEMPLATE = 'sigverify_testvectors.h.tpl'
+
+# Default output file name
+DEFAULT_OUTFILE = 'sigverify_testvectors.h'
+
+
+def compute_n0_inv(n):
+    '''Compute -(n^-1) mod 2^256, a Montgomery constant.
+
+    Sigverify expects this constant to be precomputed. Uses the standard Dusse
+    and Kaliski method; see https://eprint.iacr.org/2017/1057.pdf for more
+    details.
+    '''
+    w = 256
+    y = 1
+    # n0 = n mod 2^256
+    n0 = n & ((1 << 256) - 1)
+    # maski = 2^i - 1 (start with i=2)
+    maski = 3
+    for i in range(2, w+1):
+        # (n * y) mod 2^i = (n0 * y) mod 2^i because i <= 256
+        if (n0 * y) & maski != 1:
+            y = y + (1 << (i-1))
+        maski <<= 1
+        maski += 1
+    return (1 << w) - y
+
+
+def encode_message(msg_bytes):
+    '''Get the message encoded according to PKCS v1.5.
+
+    Returns the encoded message as big-endian bytes.
+
+    Encodes the message according to RFC 8017, section 9.2, with a SHA2-256
+    hash function. See
+    https://datatracker.ietf.org/doc/html/rfc8017#section-9.2
+    '''
+    digest = SHA256.new(msg_bytes).digest()
+
+    # T is the DER encoding of the digest
+    sha256_der_prefix = b'\x30\x31\x30\x0d\x06\x09\x60\x86\x48\x01\x65\x03\x04\x02\x01\x05\x00\x04\x20'
+    T = sha256_der_prefix + digest
```

Comment:
Any reason for the upper-case variables?

Reply:
Trying to match the notation in the RFC (specifically, step 5 [here](https://datatracker.ietf.org/doc/html/rfc8017#section-9.2)).